pilot/pkg/model/network.go

```diff
@@ -119,6 +120,9 @@ func (mgr *NetworkManager) AllGateways() []*NetworkGateway {
 	for _, gateways := range mgr.byNetwork {
 		out = append(out, gateways...)
 	}
+	sort.SliceStable(out, func(i, j int) bool {
```

Comment:
https://github.com/istio/istio/pull/34114 is trying to do the same but more comprehensively.

Reply:
Yes, please do not merge this.